module-2/backend/week_two.md

```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+    It is ORM - Object Relational Mapping tool and allows us to create, read, update and delete records from our database and much much more. 
+    
 2. What is a migration?
+    A migration is a set of instructions to build a database table. 
+    
 3. How does a table relate to a model?
+    Every line in the table is an instance of a model. 
+
 4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+    They are instance methods, I think. In case of cars/dealerships, every instance of dealerships has many cars, whereas every instance of cars belongs to a dealership. 
+    
 5. What do they allow you to do?
+    They allow us to establish relationships/associations between the objects, which in turn allow grouping and calculations. In the example above, we can group cars by dealerships and calculate statistics per dealership if needed.  
+
 6. What's the difference between agile workflow and waterfall method?
+    Waterfall is a development process in which everything is pre-planned, it is process oriented and not very flexible when changes are needed to be implemented. Agile, on the other hand, is an iterative development process, where the project is divided into measurable iterations/chunks. Agile is very flexible and allows for easy change management (I studied Business Analysis before :))
+
 7. What is the difference between `#new` and `#create`?
+    New will not save, but is a good way to look up different attributes from rails console for example. Basically: new + save = create.
+
 8. At a basic level, what does cURL allow you to do?
+    I see cURL as a tool that allows to request different URLs from terminal which otherwise would be requested from the browser.  
+
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+    I might be wrong, but I see the relationship between students/teachers as many-to-many, since one student can have many teachers available and one teacher can have many students in the class. Following this logic, we will create a Teachers-Student table which would have foreign keys (student.id, teacher.id) in it. 
+   
 10. Define foreign key, primary key, and schema.
+     Primary key is the id column value in a given table. Foreign key is a primary key from a different table which might have been added to a second table because of certain association.
+
 11. Describe the relationship between a foreign key on one table and a primary key on another table.
+     In case of cars/dealerships, the foreign key dealership.id would be placed in cars table because of one-to-many relationship between dealership and cars. 
+
 12. What are the parts of an HTTP response?
+      Method, URI, protocol, header, body. 
```

Comment:
Think these are for a request. A response will have headers and a body, and the first header will be the status code.